jdbc/src/main/java/io/kestra/jdbc/runner/JdbcExecutor.java

```diff
@@ -424,12 +426,32 @@ private void executionQueue(Either<Execution, DeserializationException> either)
                         .filter(workerTask -> this.deduplicateWorkerTask(execution, executorState, workerTask.getTaskRun()))
                         .toList();
 
+
                     // WorkerTask not flowable to workerTask
                     workerTasksDedup
                         .stream()
-                        .filter(workerTask -> workerTask.getTask().isSendToWorkerTask())
+                        .filter(workerTask -> {
+                            try {
+                                return workerTask.getTask().isSendToWorkerTask() && TruthUtils.isTruthy(workerTask.getRunContext().render(workerTask.getTask().getRunIf()));
+                            } catch (IllegalVariableEvaluationException e) {
+                                return true;
+                            }
+                        })
                         .forEach(throwConsumer(workerTask -> workerTaskQueue.emit(workerGroupService.resolveGroupFromJob(workerTask), workerTask)));
 
+                    // WorkerTask with execution condition that are not met
+                    workerTasksDedup
+                        .stream()
+                        .filter(workerTask -> {
+                            try {
+                                return !TruthUtils.isTruthy(workerTask.getRunContext().render(workerTask.getTask().getRunIf()));
+                            } catch (IllegalVariableEvaluationException e) {
+                                return false;
+                            }
+                        })
```

Comment:
This is duplicated with the above code and I wonder why you filter before and don't filter here